### PR TITLE
Add tests for certain sync features

### PR DESCRIPTION
**Pull request by wilwong89**
# Description
Tests written for version.spec.js, utils.spec.js and sync.spec.js 
Cleaned up some linting warnings for all other files

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [X] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [X] I have checked that unit tests pass locally with my changes
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by norrisng-bc — APPROVED** [review]

File: app/tests/unit/services/version.spec.js [hunk 1]
```diff
@@ -208,3 +229,34 @@ describe('update', () => {
     expect(versionTrx.commit).toHaveBeenCalledTimes(1);
   });
 });
+
+describe('updateIsLatest', () => {
+  it('Updates a version of an object', async () => {
```
Comment (jujaga) [inline app/tests/unit/services/version.spec.js:234]: This single test isn't sufficient to cover all potential branching cases. Line 310 branch isn't being fully covered: `current.isLatest` isn't being checked in the negative case.

File: app/tests/unit/services/sync.spec.js
Comment (jujaga) [inline app/tests/unit/services/sync.spec.js:1]: Upon quick inspection, this appears to only be doing formatting and indentation changes. While the coverage of syncVersions method is technically green for coverage, please double check to make sure all potential code pathways and edge cases that were modified in #225 are addressed. If needed, make sure to execute each test case independently to ensure all of the branching possibilities for this function is dealt with, as just looking at coverage when all tests are run does not provide a full picture of potentially more nuanced edge cases.

**Review by jujaga — CHANGES_REQUESTED** [review]

File: app/tests/unit/services/version.spec.js (hunk 1 above)
Comment (wilwong89) [reply app/tests/unit/services/version.spec.js]: Acceptance criteria for the ticket was limited to `versionsService.updateIsLatest()`, `utils.getUniqueObjects()`, and `syncService.syncVersions()`. Should I expand the scope of the tests?

File: app/tests/unit/services/sync.spec.js
Comment (wilwong89) [reply app/tests/unit/services/sync.spec.js:1]: I went and double checked the tests in this case, as far as I can tell the existent tests have covered all the branches and edge cases. The branch coverage stands at 80.79%, with only the try/catches with etrx/startTransaction branches remaining.

File: app/tests/unit/services/version.spec.js [hunk 2]
```diff
@@ -293,3 +296,57 @@ describe('updateIsLatest', () => {
     expect(versionTrx.commit).toHaveBeenCalledTimes(1);
   });
 });
+
+describe('RemoveDuplicateLatest', () => {
```
Comment (jujaga) [inline app/tests/unit/services/version.spec.js:300]: Style: the R in remove doesn't need to be capitalized - we usually retain the same casing as the function we're canonically testing.

File: app/tests/unit/services/version.spec.js [hunk 3]
```diff
@@ -293,3 +296,57 @@ describe('updateIsLatest', () => {
...
+  it('sets all other versions to isLatest=false', async () => {
+    listAllObjectVersionsSpy.mockResolvedValue({
+      DeleteMarkers: [{}],
+      Versions: [{ IsLatest: true, VersionId: validUuidv4 }]
+    });
+    objectSpy.mockResolvedValue({
+      path: '/test',
+      bucketId: '0000-0000'
+    });
+    Version.where.mockResolvedValueOnce([{ isLatest: true },{ isLatest: true }]);
```
Comment (jujaga) [inline app/tests/unit/services/version.spec.js:310]: Style: Double check and make sure you're right clicking and formatting with VSCode - spacing in the object array seems a bit off.

**Review by jujaga — APPROVED** [review]
Minor optional style notes.
